### PR TITLE
Update Copyright Notice

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

**Comment by crtrott** [comment]
Since this is hard to verify here is a script to make sure nothing stupid goes on: read the script and run it to see if anything weird is happening:
```
echo 'Files with contents from old copyright:'
grep -R 'Kokkos v. 2.0' | sed 's|\:| |g' | awk '{print $1}'
grep -R 'DE-AC04-94AL85000' | sed 's|\:| |g' | awk '{print $1}'
grep -R 'Copyright (2014)' | sed 's|\:| |g' | awk '{print $1}'
echo ''
echo ''

echo 'Removals other than copyright:'
git diff HEAD~1 | grep '\-' | \
  awk '{if($1=="---") {} else print $0}' |\
  awk '{if($2=="--git") {} else print $0}' |\
  awk '{if($6=="3-clause") {} else print $0}' |\
  awk '{if($7=="DE-NA0003525") {} else print $0}' |\
  awk '{if($8=="DE-NA0003525") {} else print $0}' |\
  awk '{if($1=="@@") {} else print $0}' | \
  awk '{if($2=="Copyright" && $3=="(2014)" && $4=="Sandia" && $5=="Corporation") {} else print $0}' |\
  awk '{if($7=="DE-AC04-94AL85000") {} else print $0}' |\
  awk '{if($2=="Kokkos" && $3=="v\." && $4=="2\.0") {} else print $0}' |\
  awk '{if($3=="Copyright" && $4=="(2014)" && $5=="Sandia" && $6=="Corporation") {} else print $0}' |\
  awk '{if($8=="DE-AC04-94AL85000") {} else print $0}' |\
  awk '{if($3=="Kokkos" && $4=="v\." && $5=="2\.0") {} else print $0}'

echo ''
echo ''

echo 'Additions other than copyright:'
git diff HEAD~1 | grep '\+' | \
  awk '{if($1=="+++") {} else print $0}' |\
  awk '{if($1=="@@") {} else print $0}' | \
  awk '{if($2=="Copyright" && $3=="(2020)" && $4=="National" && $5=="Technology") {} else print $0}' |\
  awk '{if($2=="Solutions" && $3=="of" && $4=="Sandia," && $5=="LLC") {} else print $0}' |\
  awk '{if($7=="DE-NA0003525") {} else print $0}' |\
  awk '{if($2=="Kokkos" && $3=="v\." && $4=="3\.0") {} else print $0}' |\
  awk '{if($8=="DE-NA0003525") {} else print $0}' |\
  awk '{if($3=="Kokkos" && $4=="v\." && $5=="3\.0") {} else print $0}'
```

File: core/src/Kokkos_Profiling_ProfileSection.hpp [hunk 1]
```diff
@@ -2,10 +2,11 @@
  //@HEADER
  // ************************************************************************
  //
- //                        Kokkos v. 2.0
- //              Copyright (2014) Sandia Corporation
+ //                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
```
Comment (masterleinad) [inline core/src/Kokkos_Profiling_ProfileSection.hpp]: This looks weird.

**Comment by masterleinad** [comment]
Retest this please.

**Comment by masterleinad** [comment]
Only the death tests are failing here.

**Comment by masterleinad** [comment]
I squashed the commits after talking to @dalg24.

**Comment by dalg24** [comment]
@crtrott what do you want to do about the CMake files with the Trilinos license?

**Comment by crtrott** [comment]
The trilinos ones stay in for now.

**Pull request merged**